mbf_abstract_nav/src/abstract_controller_execution.cpp

```diff
@@ -298,8 +298,9 @@ void AbstractControllerExecution::run()
       {
         // the specific implementation must have detected a risk situation; at this abstract level, we
         // cannot tell what the problem is, but anyway we command the robot to stop to avoid crashes
-        publishZeroVelocity();   // note that we still feedback the command calculated by the plugin
+        publishZeroVelocity();
         loop_rate_.sleep();
+        continue;
```

Comment:
Clarification question: If this safety check keeps failing, is there some mechanism that interrupts this at some point (by e.g. setting `cancel_ to` true)?

Reply:
Yeah that would be an improvement. However, the patience is currently allocated solely for the controller, so its meaning would change a bit, and the corresponding message in `isPatienceExceeded()` would need updating:

https://github.com/magazino/move_base_flex/blob/af3afbf37caa17a44ff41c71f8d1b3cc658455a2/mbf_abstract_nav/src/abstract_controller_execution.cpp#L212

Reply:
Or maybe moving the check to somewhere after checking patience and retries?

As it is clearly a bug, I'll merge this as it is by now and investigate on another PR how to react when this happens
Depending on the use case, it can be that we don't want to fail while the safety check is false for a relatively long time.
e.g. someone is using this to force the robot to wait till another agent crosses, or an elevator opens,,,

Reply:
ok sure